staging/src/k8s.io/kubectl/pkg/describe/describe.go

```diff
@@ -252,14 +269,31 @@ func DescriberFor(kind schema.GroupKind, clientConfig *rest.Config) (ResourceDes
 // GenericDescriberFor returns a generic describer for the specified mapping
 // that uses only information available from runtime.Unstructured
 func GenericDescriberFor(mapping *meta.RESTMapping, clientConfig *rest.Config) (ResourceDescriber, bool) {
+	return GenericDescriberForClient(mapping, clientConfig, nil)
+}
+
+// GenericDescriberFor returns a generic describer for the specified mapping
+// that uses only information available from runtime.Unstructured
+func GenericDescriberForClient(mapping *meta.RESTMapping, clientConfig *rest.Config, httpClient *http.Client) (ResourceDescriber, bool) {
+	var dynamicClient dynamic.Interface
+	var clientSet *clientset.Clientset
+	var err error
 	// used to fetch the resource
-	dynamicClient, err := dynamic.NewForConfig(clientConfig)
+	if httpClient == nil {
```

Comment:
if httpClient is nil, we should at least construct it once in this function and use the NewForConfigAndClient paths instead of constructing two transports here and below for the events client